start

```diff
@@ -48,14 +48,24 @@ parse_args () {
     esac
 }
 
-check_files_dont_exist () {
-    if [ -e WORKSPACE ] || [ -e BUILD ] || [ -e BazelExample.hs ]
+check_clash () {
+    if [ -f "$1" ] # directories don't clash, we're only writing files
     then
-        echo "Current directory already has WORKSPACE and/or BUILD and/or BazelExample.hs files." >&2
+        echo "ERROR: The current directory already has a $1 file we don't want to overwrite." >&2
         exit 1
     fi
 }
 
+check_files_dont_exist () {
+    # Writing BUILD.bazel does not overwrite a BUILD file but include it anyway.
+    # If we're uncomfortable with the slightly off clash message then write
+    # a separate test and message for that case.
```

Comment:
I think a separate error message would be worth the effort. The issue is that `BUILD.bazel` takes precedence and `bazel` would ignore a pre-existing `BUILD` file which may induce a fair bit of head-scratching if a user expects `bazel` to read that file.

> It avoids false positive clashes with directories such as the common build directory in many repos.

That check could also be more specific and use `[ -f` to avoid this edge-case.

For completeness we should also check for [`WORKSPACE.bazel`](https://docs.bazel.build/versions/master/build-ref.html#workspace) which would take precedence over the `WORKSPACE` file that this script generates.

Reply:
I've added checks for files that can have alternate names.